crates/core/component/distributions/src/component/rpc.rs

```diff
@@ -54,4 +54,25 @@ impl DistributionsService for Server {
             pool_size: Some(Amount::from(current_lqt_pool_size).into()),
         }))
     }
+
+    async fn lqt_pool_size_by_epoch(
+        &self,
+        request: tonic::Request<pb::LqtPoolSizeByEpochRequest>,
+    ) -> Result<tonic::Response<pb::LqtPoolSizeByEpochResponse>, tonic::Status> {
+        // Retrieve latest state snapshot.
+        let state = self.storage.latest_snapshot();
+        let epoch_index = request.into_inner().epoch;
+        let amount = state
+            .get_lqt_reward_issuance_for_epoch(epoch_index)
+            .await
+            .expect(&format!(
```

Comment:
FWIW: should avoid panicking, even if the error is isolated from the rest of the node ops. Prefer `tonic` errors with the right status, here a not found is sufficient i think (or zero, doesn't matter).